packages/devtools_app/lib/src/flutter/controllers.dart

```diff
@@ -146,6 +146,13 @@ class _ControllersState extends State<Controllers> {
   @override
   void didUpdateWidget(Controllers oldWidget) {
     super.didUpdateWidget(oldWidget);
+
+    // TODO(#1732): We create controllers twice - here, and in initState().
```

Comment:
Can we short circuit the call to _initializeProviderData in this case when we notice that oldWidget and widget have the same overrideProviders? That along with calling dispose should fix all the issues.

Reply:
I experimented with a few different things here - I'm a little less familiar with the state management going on here than I'd like to be. I'd like to leave this as a todo to revisit later, or have someone more familiar with this area take a look.